src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1215,29 +1234,531 @@ public static VariantContext makeFromAlleles(final String name, final String con
+            }
+
+            return biallelics;
+        }
+    }
+
+    /**
+     * Add the VCF INFO field annotations for the used alleles
+     *
+     * @param vc                    original variant context
+     * @param builder               variant context builder with subset of original variant context's alleles
+     * @param keepOriginalChrCounts keep the orignal chromosome counts before subsetting
+     * @return variant context builder with updated INFO field attribute values
+     */
+    private static void addInfoFieldAnnotations(final VariantContext vc, final VariantContextBuilder builder,
+                                                final boolean keepOriginalChrCounts) {
+        Utils.nonNull(vc);
+        Utils.nonNull(builder);
+        Utils.nonNull(builder.getAlleles());
+
+        final List<Allele> alleles = builder.getAlleles();
+        if (alleles.size() < 2)
+            throw new IllegalArgumentException("the variant context builder must contain at least 2 alleles");
+
+        // don't have to subset, the original vc has the same number and hence, the same alleles
+        boolean keepOriginal = (vc.getAlleles().size() == alleles.size());
+
+        List<Integer> alleleIndecies = builder.getAlleles().stream().map(a -> vc.getAlleleIndex(a)).collect(Collectors.toList());
+        if (keepOriginalChrCounts) {
+            if (vc.hasAttribute(VCFConstants.ALLELE_COUNT_KEY))
+                builder.attribute(GATKVCFConstants.ORIGINAL_AC_KEY, keepOriginal ?
+                        vc.getAttribute(VCFConstants.ALLELE_COUNT_KEY) : alleleIndecies.stream().filter(i -> i > 0).map(j -> vc.getAttributeAsList(VCFConstants.ALLELE_COUNT_KEY).get(j - 1)).collect(Collectors.toList()).get(0));
+            if (vc.hasAttribute(VCFConstants.ALLELE_FREQUENCY_KEY))
+                builder.attribute(GATKVCFConstants.ORIGINAL_AF_KEY, keepOriginal ?
+                        vc.getAttribute(VCFConstants.ALLELE_FREQUENCY_KEY) : alleleIndecies.stream().filter(i -> i > 0).map(j -> vc.getAttributeAsList(VCFConstants.ALLELE_FREQUENCY_KEY).get(j - 1)).collect(Collectors.toList()).get(0));
+            if (vc.hasAttribute(VCFConstants.ALLELE_NUMBER_KEY)) {
+                builder.attribute(GATKVCFConstants.ORIGINAL_AN_KEY, vc.getAttribute(VCFConstants.ALLELE_NUMBER_KEY));
+            }
+        }
+
+        VariantContextUtils.calculateChromosomeCounts(builder, true);
+    }
+
+    /**
+     * Subset the Variant Context to the specific set of alleles passed in (pruning the PLs appropriately)
+     *
+     * @param vc              variant context with genotype likelihoods
+     * @param allelesToUse    which alleles from the vc are okay to use; *** must be in the same relative order as those in the original VC ***
+     * @param assignGenotypes assignment strategy for the (subsetted) PLs
+     * @return a new non-null GenotypesContext with subsetted alleles
+     */
+    public static GenotypesContext subsetAlleles(final VariantContext vc,
```

Comment:
This method should be replaced by the existing codepath in AlleleSubsettingUtils